### PR TITLE
Add tests for ucsd_features and modify existing tests accordingly

### PR DESCRIPTION
**Pull request by danialmalik**
### Description
Add tests for UCSD features and customizations.

### How to test
```
paver test_system -s lms -t lms/djangoapps/commerce/tests/test_utils.py::RefundUtilMethodTests::test_ecommerce_refund_failed_process_notification_email_sent  --disable-migrations
paver test_system -s lms -t openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py::TestUtils::test_send_email_instead_zendesk_ticket_successfully --disable-migrations
paver test_system -s lms -t openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py::TestUtils::test_send_email_instead_zendesk_ticket_failed --disable-migrations
paver test_system -s lms -t openedx/features/ucsd_features/tests/test_utils.py --disable-migrations
```

### CODE REVIEW

File: openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 import ddt
-from django.test.utils import override_settings
 from mock import MagicMock, patch
 
+from django.test.utils import override_settings
+from django.conf import settings
```
Comment (mumarkhan999) [inline openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py]: SORT THE IMPORTS.

File: lms/djangoapps/commerce/tests/test_utils.py [hunk 2]
```diff
@@ -269,6 +269,37 @@ def test_ecommerce_refund_failed_process_notification_sent(self, mock_send_notif
         assert call_args[0] == (course_entitlement.user, [1])
         assert refund_success
 
+    @httpretty.activate
+    @patch('lms.djangoapps.commerce.utils.send_notification_email_to_support', return_value=True)
+    def test_ecommerce_refund_failed_process_notification_email_sent(self, mock_send_notification):
+        """
+        Test that email is sent for notification instead of creating a ticket on zendesk when
+        `ENABLE_EMAIL_INSTEAD_ZENDESK` feature flag is set to `True`
```
Comment (mumarkhan999) [inline lms/djangoapps/commerce/tests/test_utils.py]: Kindly have a look at the docstrings again. I think these are a little ambiguous.

File: lms/djangoapps/commerce/tests/test_utils.py [hunk 3]
```diff
@@ -269,6 +269,37 @@ def test_ecommerce_refund_failed_process_notification_sent(self, mock_send_notif
         assert call_args[0] == (course_entitlement.user, [1])
         assert refund_success
 
+    @httpretty.activate
+    @patch('lms.djangoapps.commerce.utils.send_notification_email_to_support', return_value=True)
+    def test_ecommerce_refund_failed_process_notification_email_sent(self, mock_send_notification):
```
Comment (mumarkhan999) [inline lms/djangoapps/commerce/tests/test_utils.py]:
```suggestion
    def test_ecommerce_refund_failed_process_notification_email_sent(self, mocked_send_notification):
```

File: openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py [hunk 4]
```diff
@@ -56,3 +59,57 @@ def test_unexpected_error_pinging_zendesk(self):
...
+                mocked_send_notification.call_args[1]['message_type'],
+                notification_message_type
+            )
+            self.assertEqual(return_value, status.HTTP_201_CREATED)
+
+    @patch('openedx.core.djangoapps.zendesk_proxy.utils.send_notification_email_to_support', return_value=False)
+    def test_send_email_instead_zendesk_ticket_failed(self, mocked_send_notification):
+        """
+        Test that email is sent for notification instead of creating a ticket on zendesk when
+        `ENABLE_EMAIL_INSTEAD_ZENDESK` feature flag is set to `True`. If email is not send successfully
```
Comment (mumarkhan999) [inline openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py]:
```suggestion
        `ENABLE_EMAIL_INSTEAD_ZENDESK` feature flag is set to `True`. If email is not sent successfully,
```

File: openedx/features/ucsd_features/tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+from django.conf import settings
+from django.test import TestCase
+
+from edx_ace.recipient import Recipient
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.features.ucsd_features.utils import send_notification, send_notification_email_to_support
+from openedx.features.ucsd_features.message_types import ContactSupportNotification, CommerceSupportNotification
```
Comment (mumarkhan999) [inline openedx/features/ucsd_features/tests/test_utils.py]: Sort the imports.

File: openedx/features/ucsd_features/tests/test_utils.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        """
+        returned_value = send_notification(message_type, {}, self.test_emails)
+
+        mocked_send.assertCalled()
+        self.assertFalse(returned_value)
+
+    @patch('openedx.features.ucsd_features.utils.send_notification', return_value=True)
+    def test_send_notification_email_to_support_method(self, mocked_send_notification):
+        """
+        Test the send_notification_email_to_support method.
```
Comment (mumarkhan999) [inline openedx/features/ucsd_features/tests/test_utils.py]: Kindly use some other meaningful docstring for this one like you did above.

File: lms/djangoapps/commerce/tests/test_utils.py (hunk 2 above)
Comment (danialmalik) [reply lms/djangoapps/commerce/tests/test_utils.py]: Do you have something in mind? I can't think of something simpler.

File: lms/djangoapps/commerce/tests/test_utils.py (hunk 3 above)
Comment (danialmalik) [reply lms/djangoapps/commerce/tests/test_utils.py]: I am trying to be consistent with existing Open edX tests. In the previous test, they have used `mock_send_notification` and so have I.

File: openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py (hunk 1 above)
Comment (tehreem-sadat) [reply openedx/core/djangoapps/zendesk_proxy/tests/test_utils.py]: done.

**Pull request merged**